pyproject.toml

```diff
@@ -1,91 +1,137 @@
-[tool.poetry]
-name = "LISA"
-version = "3.0.0.dev1"
+[build-system]
+requires = ["setuptools >= 65", "setuptools_scm[toml] >= 6.2"]
+build-backend = "setuptools.build_meta"
+
+
+[project]
+name = "lisa"
 description = "Linux Integration Services Automation"
-authors = ["contributors <https://github.com/microsoft/lisa/graphs/contributors>"]
-license = "MIT"
-packages = [
-    { include = "lisa" },
-    { include = "microsoft" },
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Development Status :: 5 - Production/Stable",
+    "Topic :: Software Development :: Testing",
+]
+dependencies = [
+    "assertpy ~= 1.1",
+    "func-timeout ~= 4.3.5",
+    "dataclasses-json ~= 0.5.2",
+    "paramiko ~= 2.10.1",
+    "pluggy ~= 0.13.1",
+    "python-dateutil ~= 2.8.1",
+    "pytest-html ~= 3.1.1",
+    "PyYAML ~= 5.4.1",
+    "randmac ~= 0.1",
+    "retry ~= 0.9.2",
+    "semver ~= 2.13.0",
+    "spurplus ~= 2.3.4",
+    "websockets ~= 10.3",
+]
+dynamic = ["version"]
+license = {text = "MIT"}
+readme = "README.rst"
+requires-python = ">=3.8"
+
+
+[project.optional-dependencies]
+aws = [
+    "boto3 ~= 1.21.37",
+]
+
+azure = [
+    "azure-identity ~= 1.9.0",
+    "azure-mgmt-compute ~= 26.1.0",
+    "azure-mgmt-marketplaceordering ~= 1.1.0",
+    "azure-mgmt-network ~= 19.3.0",
+    "azure-mgmt-privatedns ~= 1.0.0",
+    "azure-mgmt-resource ~= 21.0.0",
+    "azure-mgmt-serialconsole ~= 1.0.0",
+    "azure-mgmt-storage ~= 20.0.0",
+    "azure-storage-blob ~= 12.11.0",
+    "azure-storage-file-share ~= 12.4.0",
+    "cachetools ~= 5.2.0",
+    "requests",
+    "Pillow ~= 9.0.0",
+    "PyGObject ~= 3.38.0; platform_system == 'Linux'",
+]
+
+docs = [
+    "Sphinx >= 4.1.0",
+    "sphinx-argparse >= 0.2.5",
+    "sphinx-rtd-theme >= 0.5.2",
+    "sphinxemoji >= 0.1.8",
+    "sphinx-copybutton >= 0.4.0",
+]
+
+legacy = [
+    "pypiwin32; platform_system == 'Windows'",
+]
+
+libvirt = [
```

Comment:
libvirt works in Linux only, how to avoid to install the whole things on Windows?

Reply:
These are extras, so if you want libvirt support you'd run `pip install .[libvirt]`. The Linux docs have `[azure, libvirt]` and Windows docs have `[azure]`. Probably the extras we want other to consume should probably get documented. This lets a user install only the dependencies that they need.

Reply:
If we don't explicitly set the libvirt on Linux only, people will experience failure on Windows, confused, and file issues.